splink/em_training_session.py

```diff
@@ -165,6 +166,20 @@ def _train(self):
 
         cvv = self._comparison_vectors()
 
+        # check that the blocking rule actually generates _some_ record pairs,
+        # if not give the user a helpful message
+        if not cvv.as_record_dict(limit=1):
```

Comment:
Just to note: I wanted to check on the performance of this.  It's a slightly expensive operation especially in Athena and Spark, because it requires a query to be executed.

I think I'm happy with it on the basis that the the comparison vectors are cached, so it's a trivial query, but there's still some overhead, especially in Athena.

The other option would be to attempt to catch the `nan` error, but I think that in itself would be tricky, so I'm happy with this.

Mainly just writing this down so you know what I looked at to check this was ok

Reply:
I guess if this did become a sticking point, another option (perhaps at the cost of some user-friendliness) would be to allow the 'advanced' user to pass a flag which allows skipping this check, on the basis that they should know what they're doing sufficiently